modules/contact/src/actions/ContactAction.C

```diff
@@ -249,6 +249,51 @@ ContactAction::ContactAction(const InputParameters & params)
   else if (getParam<bool>("ping_pong_protection"))
     paramError("ping_pong_protection",
                "The 'ping_pong_protection' option can only be used with the 'ranfs' formulation");
+
+  // Remove repeated pairs from input file.
+  removeRepeatedPairs();
+}
+
+void
+ContactAction::removeRepeatedPairs()
+{
+  if (_boundary_pairs.size() == 0)
+    paramError(
+        "primary",
+        "Number of contact pairs in the contact action is zero. Please revise your input file.");
+
+  // Remove repeated interactions
+  std::vector<std::pair<BoundaryName, BoundaryName>> lean_boundary_pairs;
+
+  for (const auto & [primary, secondary] : _boundary_pairs)
+  {
+    const auto & primary_lambda = primary;
+    const auto & secondary_lambda = secondary;
+
+    auto it = std::find_if(lean_boundary_pairs.begin(),
+                           lean_boundary_pairs.end(),
+                           [&](const std::pair<BoundaryName, BoundaryName> & lean_pair)
+                           {
+                             const bool match_one = lean_pair.second == secondary_lambda &&
+                                                    lean_pair.first == primary_lambda;
+                             const bool match_two = lean_pair.second == primary_lambda &&
+                                                    lean_pair.first == secondary_lambda;
+                             const bool exist = match_one || match_two;
+                             return exist;
+                           });
+
+    if (it == lean_boundary_pairs.end())
+      lean_boundary_pairs.push_back({primary, secondary});
```

Comment:
Why destructure it above if you just need to reassemble it down here? might as well keep it as `pair` and use `pair.first` instead of `primary_lambda`... Maybe I'm missing something here, but this just looks odd

Reply:
I had it like that initially, but I thought it was somewhat clearer to use primary/secondary.

Reply:
I can change it if you like, but I believe it's clear enough as it is.

Reply:
Maybe @dschwen would like it better conceptually with a 
```suggestion
      lean_boundary_pairs.emplace_back(primary, secondary);
```

Reply:
Made that change in latest push.